core/chains/evm/client/errors.go

```diff
@@ -444,65 +481,70 @@ func ExtractRPCError(baseErr error) (*JsonError, error) {
 	return &jErr, nil
 }
 
-func ClassifySendError(err error, lggr logger.SugaredLogger, tx *types.Transaction, fromAddress common.Address, isL2 bool) commonclient.SendTxReturnCode {
+func ClassifySendError(err error, clientErrors config.ClientErrors, lggr logger.SugaredLogger, tx *types.Transaction, fromAddress common.Address, isL2 bool) commonclient.SendTxReturnCode {
 	sendError := NewSendError(err)
 	if sendError == nil {
 		return commonclient.Successful
 	}
-	if sendError.Fatal() {
+
+	configErrors := ClientErrorRegexes(clientErrors)
+
+	if sendError.Fatal() || configErrors.ErrIs(sendError, Fatal) {
 		lggr.Criticalw("Fatal error sending transaction", "err", sendError, "etx", tx)
 		// Attempt is thrown away in this case; we don't need it since it never got accepted by a node
 		return commonclient.Fatal
 	}
-	if sendError.IsNonceTooLowError() || sendError.IsTransactionAlreadyMined() {
+	if sendError.IsNonceTooLowError() || sendError.IsTransactionAlreadyMined() ||
+		configErrors.ErrIs(sendError, NonceTooLow, TransactionAlreadyMined) {
 		lggr.Debugw(fmt.Sprintf("Transaction already confirmed for this nonce: %d", tx.Nonce()), "err", sendError, "etx", tx)
 		// Nonce too low indicated that a transaction at this nonce was confirmed already.
 		// Mark it as TransactionAlreadyKnown.
 		return commonclient.TransactionAlreadyKnown
 	}
-	if sendError.IsReplacementUnderpriced() {
+	if sendError.IsReplacementUnderpriced() || configErrors.ErrIs(sendError, ReplacementTransactionUnderpriced) {
```

Comment:
These seem to be tangling up different layers and replicating things. Can we align them one way or another? For example:
```suggestion
	if sendError.IsReplacementUnderpriced(configErrors) {
```

Reply:
Yeah for sure, I was debating on passing the configErrors like this actually.